### PR TITLE
Feature/get mappings

### PR DESCRIPTION
**Pull request by hugowetterberg**
Makes it possible for clients (and developers) to inspect what the documents get mapped to in Open Search. The `type` is mostly the same as the actual OS type for the property, but the goal is to expose the "logical" type, not details about the OS implementation. That's why the actual mapping:

``` json
  "document.title": {
    "type": "text",
    "fields": {
      "prefix": {
        "type": "text",
        "analyzer": "elephant_prefix_analyzer",
        "search_analyzer": "elephant_prefix_search_analyzer"
      }
    }
  },
```

Gets translated to:

```
    {
      "name": "document.title",
      "type": "text",
      "fields": [
        {
          "name": "prefix",
          "type": "prefix"
        }
      ]
    },
```

I plan to follow this up with a second PR that a) changes the indexer so that all the properties that have been `keyword` use the `icu_collation_keyword` type instead. When that happens I think that they just should be reported as `keyword`, as that's what they are, they're just in the language one would expect from the document language.

### CODE REVIEW

**Comment by hugowetterberg** [comment]
Implementation of https://github.com/ttab/elephant-api/pull/20

**Comment by hugowetterberg** [comment]
I added the icu-change to this PR, dealing with chained PR:s is so messy.

**Comment by hugowetterberg** [comment]
The individual changesets are still readable in atomic commits https://github.com/ttab/elephant-index/pull/139/commits/fd7e41fe778a8f0cebbec1ff1b8551bd03041587 (GetMappingSource) and https://github.com/ttab/elephant-index/pull/139/commits/a9f09b96ee653b25e0c910e4929deebcb48dbfbd (use icu_collation_keyword),

File: index/server.go [hunk 1]
```diff
@@ -85,7 +85,10 @@ func RunIndex(ctx context.Context, p Parameters) error {
 	registerAPI(router, opts, api)
 
 	seachAPI := index.NewSearchV1Server(
```
Comment (danijelvukoje) [inline index/server.go:87]: Typo? `searchAPI`

File: index/search_api.go [hunk 2]
```diff
@@ -43,21 +53,117 @@ func NewSearchServiceV1(
...
+
+	return "", false
+}
+
 // Query implements index.SearchV1.
 func (s *SearchServiceV1) Query(
 	ctx context.Context, req *index.QueryRequestV1,
 ) (_ *index.QueryResponseV1, outErr error) {
-	auth, err := RequireAnyScope(ctx, ScopeSeach)
+	auth, err := RequireAnyScope(ctx, ScopeSeach, ScopeIndexAdmin)
```
Comment (danijelvukoje) [inline index/search_api.go:168]: Typo? `ScopeSearch`

**Review by danijelvukoje — CHANGES_REQUESTED** [review]

**Review by danijelvukoje — APPROVED** [review]
Sweet! 🙂

**Pull request merged**